client/blocks/reader-import-button/index.jsx

```diff
@@ -0,0 +1,86 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { noop } from 'lodash';
+import { localize } from 'i18n-calypso';
+import Gridicon from 'gridicons';
+
+/**
+ * Internal dependencies
+ */
+import wpcom from 'lib/wp';
+import FilePicker from 'components/file-picker';
+
+class ReaderImportButton extends React.Component {
+
+	static propTypes = {
+		onError: React.PropTypes.func,
+		onImport: React.PropTypes.func,
+		onProgress: React.PropTypes.func,
+	}
+
+	static defaultProps = {
+		onError: noop,
+		onImport: noop,
+		onProgress: noop,
+	}
+
+	state = { disabled: false };
+
+	onClick = ( event ) => {
+		// Don't allow picking of a new file if there's an import in progress
+		if ( this.state.disabled ) {
+			event.preventDefault();
+		}
+	}
+
+	onPick = ( files ) => {
+		// We only care about the first file in the list
+		const file = files[ 0 ];
+		if ( ! file ) {
+			return;
+		}
+
+		this.fileName = file.name;
+		const req = wpcom.undocumented().importReaderFeed( file, this.onImport );
```

Comment:
do you think this is something we will eventually want to move to redux? I'm guessing its a holdover from the earliest days?